src/Daemon/BaseDaemon.cpp

```diff
@@ -154,7 +154,10 @@ static void signalHandler(int sig, siginfo_t * info, void * context)
     writePODBinary(*info, out);
     writePODBinary(signal_context, out);
     writePODBinary(stack_trace, out);
-    writeVectorBinary(Exception::thread_frame_pointers, out);
+    if (Exception::enable_job_stack_trace)
+        writeVectorBinary(Exception::thread_frame_pointers, out);
+    else
+        writeVarUInt(0, out);
```

Comment:
woudn't `writeVectorBinary(std::vector<StackTrace::FramePointers>{}, out);` be a cleaner way?

Reply:
it will end up with the same problem - initializing std::vector which involves calling to malloc which is signal-unsafe

Reply:
Hmm.. I thought that empty vector doesn't need malloc

Reply:
https://godbolt.org/z/YaM43Mv6M